### ISSUE TITLE
Omit empty as an option

### ISSUE THREAD

**Issue by johngb** [davecgh/go-spew#22]
Often I'm working with large structs which have fields which are usually empty, but which are of a lot of interest if not empty.  I would love there to be an option to omit empty fields.  The problem of course is how we define empty.

I would suggest that simply considering the zero or nil state of any field to be an empty field.  So, "" would be empty for a string, and 0 would be empty for an int.

**Comment by davecgh** [comment]
This is a reasonable option I think.  It would be based off the zero value, imo.

**Comment by jawspeak** [comment]
Any update or pending PR anyone has been working on this? Would be helpful to omit nil / zero values. Is there a test you'd suggest I look at 1st to get to speed with implementing this?

**Comment by davecgh** [comment]
I don't believe there is anyone working on it.  I'm only in maintenance mode on this at the moment to ensure continues to work as Go is updated.

That said, I would accept a pull request that provides an option for this as I think it could certainly be useful.  Spew has extremely thorough tests at the moment (100% coverage across all Go versions), so I think all you'd basically want to do is add tests with the zero value for each of the types there are already there to ensure they are omitted when the option is set.

**Comment by ghostsquad** [comment]
just stumbled on wanting to do this exact thing.
